dao/src/main/java/com/carhouse/dao/mappers/CarSaleMapper.java

```diff
@@ -45,21 +54,39 @@
     /**
      * Instantiates a new Car sale mapper.
      *
-     * @param userMapper               the user mapper
-     * @param carMapper the car mapper
+     * @param userMapper    the user mapper
+     * @param carMapper     the car mapper
+     * @param commentMapper the comment mapper
      */
     @Autowired
-    public CarSaleMapper(final UserMapper userMapper, final CarMapper carMapper) {
+    public CarSaleMapper(final UserMapper userMapper, final CarMapper carMapper, final CommentMapper commentMapper) {
         this.userMapper = userMapper;
         this.carMapper = carMapper;
+        this.commentMapper = commentMapper;
     }
 
+    /**
+     * Auto increment starts from 1, so if comment id 0 it means what comments do not exist end created by left join.
+     * @param resultSet     resultSet
+     * @param i             row number
+     * @return              carSale object
+     * @throws SQLException exception
+     */
     @Override
     public CarSale mapRow(final ResultSet resultSet, final int i) throws SQLException {
         CarSale carSale = new CarSale(resultSet.getInt(CAR_SALE_ID), resultSet.getBigDecimal(PRICE),
                 resultSet.getDate(DATE), userMapper.mapRow(resultSet, i), carMapper.mapRow(resultSet, i));
-        LOGGER.debug("row ({}, {}, {}) has been mapped", carSale.getCarSaleId(), carSale.getPrice(),
-                carSale.getDate());
+        List<Comment> commentList = new ArrayList<>();
+        Comment comment;
+        do {
```

Comment:
Personally for me do-while is usually risky in case when resultset doesn't have any rows for mapping